### PR TITLE
chore: Refactor Terraform module to follow spec CC006

### PR DESCRIPTION
**Pull request by ghislainbourgeois**
# Description

Refactor the Terraform module to follow spec CC006.

The integration in particular have changed, and will require other modules to be changed at the same time. This is set to draft until all the PRs are ready.

# Checklist:

- [x] My code follows the [style guidelines](/CONTRIBUTING.md) of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that validate the behaviour of the software
- [x] I validated that new and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have bumped the version of the library

### CODE REVIEW

**Review by gruyaume — APPROVED** [review]
Looks good, do you want to move this to draft to prevent anybody from merging it pre-preemptively?

File: terraform/README.md [hunk 1]
```diff
@@ -16,18 +16,22 @@ rather serve as a building block for higher level modules.
...
+- **versions.tf** - Defines the Terraform provider version.
 - 
 ## Using sdcore-amf-k8s base module in higher level modules
 
 If you want to use `sdcore-amf-k8s` base module as part of your Terraform module, import it
 like shown below:
 
 ```text
+data "juju_model" "my_model" {
+  name = "my_model_name"
```
Comment (Gmerold) [inline terraform/README.md:28]: Shouldn't we use `var.model` here? And then, in line 43, use the reference to the object just like we're doing in line 34?

File: terraform/README.md [hunk 2]
```diff
@@ -36,14 +40,14 @@ Create integrations, for instance:
 
 ```text
 resource "juju_integration" "amf-nrf" {
-  model = var.model_name
+  model = var.model
```
Comment (Gmerold) [inline terraform/README.md:43]: Following the previous comment, why not use `juju_model.my_model.name`?

**Review by Gmerold — CHANGES_REQUESTED** [review]

File: terraform/README.md (hunk 2 above)
Comment (ghislainbourgeois) [reply terraform/README.md]: Done, thanks!

**Review by Gmerold — APPROVED** [review]

**Pull request merged**